src/rpc/net.cpp

```diff
@@ -968,6 +968,53 @@ static RPCHelpMan addpeeraddress()
     };
 }
 
+static RPCHelpMan sendmsgtopeer()
+{
+    return RPCHelpMan{"sendmsgtopeer",
+                "\nSend a p2p message to a peer specified by id.\n"
+                "The message type and body must be provided, the message header will be generated"
+                "This RPC is for testing only\n",
+                {
+                    {"peer_id", RPCArg::Type::NUM, RPCArg::Optional::NO, "The peer to send the message to."},
+                    {"msg_type", RPCArg::Type::STR, RPCArg::Optional::NO, strprintf("The message type (maximum length %i)", CMessageHeader::COMMAND_SIZE)},
+                    {"msg", RPCArg::Type::STR_HEX, RPCArg::Optional::NO, "The serialized message body to send, in hex, without a message header"},
+                },
+                RPCResult{RPCResult::Type::OBJ, "", "", std::vector<RPCResult>{}},
+                RPCExamples{
+                    HelpExampleCli("sendmsgtopeer", "0 \"addr\" \"ffffff\"")
+            + HelpExampleRpc("sendmsgtopeer", "0 \"addr\" \"ffffff\"")
+                },
+        [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
+{
+    const NodeId peer_id{request.params[0].getInt<int64_t>()};
+    const std::string& msg_type{request.params[1].get_str()};
+    if( msg_type.size() > CMessageHeader::COMMAND_SIZE) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Error: msg_type too long, max is (%i)", CMessageHeader::COMMAND_SIZE));
+    }
+    const std::vector<unsigned char> msg{ParseHex(request.params[2].get_str())};
+
+    NodeContext& node = EnsureAnyNodeContext(request.context);
+    CConnman& connman = EnsureConnman(node);
+
+    CSerializedNetMsg msg_ser;
+    msg_ser.data = msg;
+    msg_ser.m_type = msg_type;
+
+    bool success = connman.ForNode(peer_id, [&](CNode* node) {
+        connman.PushMessage(node, std::move(msg_ser));
+        return true;
+    });
+
+    if (!success) {
+        throw JSONRPCError(RPC_MISC_ERROR, "Error: Could not send message to peer");
+    }
+
+    UniValue ret{UniValue::VOBJ};
+    return ret;
+},
+    };
+}
```

Comment:
For some reason I always thought that clang-format doesn't apply to rpc indentation (because most of the existing ones don't comply with it), but that doesn't seem to be the case, so I've applied it.